addons/l10n_ae/models/invoice.py

```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models, fields
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    # Technical field to show rate on invoice report
+    l10n_ae_invoice_rate = fields.Float(compute='_compute_rate', string='Rate on Invoice Date', digits=(12, 6))
```

Comment:
@rgo-odoo , I think  l10n_ae_invoice_rate should be storable. 
If someone update currency rate of that period, should not affected on old invoices

cc @jco-odoo

Reply:
@jco-odoo , right now, there is not any restriction to update currency rate of the past.

I do not think that we should put restriction.

Reply:
I don't think we should store any of this, or have any extra fields. We already have a history of currency rates and we can do the conversions when we print the invoice. Any reason why we need any extra fields at all?

Reply:
@odony right, there is no need to add extra fields, it can be managed from report itself like you suggested. Also removed restriction for text `Tax` in invoice for partner's GCC countries.
Thanks.